src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,180 @@
+            lookUpNode.prev.next = newNode;
+            newNode.prev = lookUpNode.prev;
+            newNode.next = lookUpNode;
+            lookUpNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bounds: 0 - " + size);
+        }
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bounds: 0 - " + size);
+        } else {
+            T removedObject = findNodeByIndex(index).item;
+            findNodeByIndex(index).item = value;
+            return removedObject;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        verifyIndex(index);
+        if (index == size) {
+            throw new IndexOutOfBoundsException();
+        }
+        T value = findNodeByIndex(index).item;
+        Node<T> lookUpNode = findNodeByIndex(index);
+        if (index == 0) {
+            if (size == 1) {
+                lookUpNode.item = null;
+                size--;
+                return value;
+            }
+            lookUpNode.next.prev = null;
+            head = lookUpNode.next;
+            size--;
+            return value;
+        }
+        if (index == size - 1) {
+            lookUpNode.prev.next = null;
+            tail = lookUpNode.prev;
+            size--;
+            return value;
+        } else {
+            lookUpNode.prev.next = lookUpNode.next;
+            lookUpNode.next.prev = lookUpNode.prev;
+            size--;
+            return value;
+        }
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = findNodeIndexByItem(object);
+        if (index < 0) {
+            return false;
+        } else {
+            remove(index);
+            return true;
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void verifyIndex(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bounds: 0 - " + size);
+        }
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        verifyIndex(index);
+        Node<T> lookUpNode = head;
+        for (int i = 0; i < index; i++) {
+            lookUpNode = lookUpNode.next;
+        }
+        return lookUpNode;
+    }
+
+    private Node<T> findNodeByItem(T lookUpItem) {
```

Comment:
Where do you use this method?

Reply:
nowhere:) i make remove(T object) method in different way, forgot to delete this old method